miqi/runtime/context_runtime.py

```
@@ -396,7 +470,13 @@ def trim_for_model(
             "Pre-send guard: messages {} -> {} (est tokens {} -> {})",
             len(messages), len(work), est, est_after,
         )
-        return work
+        # 结构修复：trim 的组删除受「保留最后一条」保护，可能删掉
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)